### PR TITLE
add security configuration to secure management API with an API key

### PR DESCRIPTION
**Pull request by PierreBesson**

### CODE REVIEW

File: jhipster-framework/src/main/java/io/github/jhipster/security/management/ApiKeyAuthenticationFilter.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import org.springframework.web.filter.GenericFilterBean;
+
+/**
+ * Filters incoming requests and install the "manager" Spring Security principal if the Authorization header
+ * contains the correct API Key.
+ */
+public class ApiKeyAuthenticationFilter extends GenericFilterBean {
+    static final String AUTHORIZATION_HEADER = "Authorization";
+    static final String MANAGER = "manager";
+    static final String ROLE_MANAGEMENT = "ROLE_MANAGEMENT";
```
Comment (cbornet) [inline jhipster-framework/src/main/java/io/github/jhipster/security/management/ApiKeyAuthenticationFilter.java]: Do we need a new role ? Couldn't it just be an ADMIN API key ?

File: jhipster-framework/src/main/java/io/github/jhipster/security/management/ApiKeyAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.GenericFilterBean;
+
+/**
+ * Filters incoming requests and install the "manager" Spring Security principal if the Authorization header
+ * contains the correct API Key.
+ */
+public class ApiKeyAuthenticationFilter extends GenericFilterBean {
+    static final String AUTHORIZATION_HEADER = "Authorization";
+    static final String MANAGER = "manager";
```
Comment (cbornet) [inline jhipster-framework/src/main/java/io/github/jhipster/security/management/ApiKeyAuthenticationFilter.java]: This user doesn't really exist (eg. in DB). I find it weird...

File: jhipster-framework/src/main/java/io/github/jhipster/security/management/ApiKeyAuthenticationFilter.java (hunk 1 above)
Comment (PierreBesson) [reply jhipster-framework/src/main/java/io/github/jhipster/security/management/ApiKeyAuthenticationFilter.java]: Well my idea, was maybe you don't want to give full ADMIN rights, just the rights to access /management/*..

File: jhipster-framework/src/main/java/io/github/jhipster/security/management/ApiKeyAuthenticationFilter.java (hunk 2 above)
Comment (PierreBesson) [reply jhipster-framework/src/main/java/io/github/jhipster/security/management/ApiKeyAuthenticationFilter.java]: I agree that it's strange but then this user need a name.

**Comment by PierreBesson** [comment]
Closing as we decided to go for the basic auth solution.

**Pull request closed without merging**